File: csharp/src/AutomationRoot/Automation.Testing/Cases/StudentDetails.cs
using Automation.Api.Pages;
using Automation.Core.Testing;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Automation.Testing.Cases
{
    public class StudentDetails : TestCase
    {
        public override bool AutomationTest(IDictionary<string, object> testParams)
        {
            // students to find
            var keyword = $"{testParams["keyword"]}";
            var fluent = $"{testParams["fluent"]}";
            var students = $"{testParams["students"]}";

            // perform test case
            var student = CreateFluentApi(fluent)
                .ChangeContext<IStudents>(students, $"{testParams["application"]}")
                .FindByName(keyword)
                .Students()
                .First();

            // extract expected
            var expected = student.FirstName();

            // assert
            return student.Details().FirstName().Equals(expected, StringComparison.OrdinalIgnoreCase);
        }
    }
}

File: java/src/automation-root/automation-testing/src/test/java/com/automation/testing/cases/StudentDetails.java
package com.automation.testing.cases;

import com.automation.api.components.Student;
import com.automation.api.pages.Students;
import com.automation.core.testing.TestCase;

import java.io.IOException;
import java.lang.reflect.InvocationTargetException;
import java.util.Map;

public class StudentDetails extends TestCase {
    @Override
    public boolean automationTest(Map<String, Object> testParams)
            throws IOException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {

        // create driver for this test case
        String application = testParams.get("application").toString();
        String keyword = testParams.get("keyword").toString();
        String fluent = testParams.get("fluent").toString();
        String students = testParams.get("students").toString();

        Student student = newFluentApi(fluent)
                .<Students>changeContext(students, application)
                .findByName(keyword)
                .students()
                .stream()
                .findFirst()
                .orElse(null);

        // extract expected result
        assert student != null;
        String expected = student.firstName();

        // assert
        return student.details().firstName().equals(expected);
    }
}